dhttp/server.go

```diff
@@ -1,3 +1,35 @@
+// Package dhttp is a simple production-ready HTTP server library for the 2020s.
```

Comment:
The docs rock. I'm going to comment some random nits, but I want to open with the fundamental rockingness here. 👍